### DISCUSSION TITLE
The Trainer freeze when I use the logger

### DISCUSSION THREAD

**Discussion by xugaoqi1993** [Lightning-AI/pytorch-lightning#16374] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hello:
I am training my model with pytorch lightning. If I set the logger as False there are no issues. However, if i use the logger, the training processing will freeze and no error will be reported. Here are the screenprints when this problem occurs. The gpu memory will be occupied by about 1700mb, which is abount 23000mb in a normal training processing. It seems like the training process is stuck when the data is transported into gpu memory. 
Is there any solutions to slove this problems?
Thank you
<img width="1785" alt="1" src="https://user-images.githubusercontent.com/70183461/212613893-6ce54888-1121-457f-b678-a0c84345316c.png">
<img width="1790" alt="1" src="https://user-images.githubusercontent.com/70183461/212613955-305030e3-d5a1-4bf5-8e4c-5eb2f31e0720.png">
<img width="1000" alt="1" src="https://user-images.githubusercontent.com/70183461/212614648-79f93022-2946-451d-8505-a3b1984a49d4.png">

**Comment by CindyZJT** [comment]
I'm having the same issue too. Did you solve it ?

**Reply by jxtngx** [reply]
Hi @CindyZJT. Can you please tell me which version of Lightning you are using. And which logger.

**Reply by CindyZJT** [reply]
I'm using pl with version 2.0.6. 
I set trainer like this.
<img width="732" alt="image" src="https://github.com/Lightning-AI/lightning/assets/17687676/bbb89a86-9e7f-45fb-838b-c16588ce9920">

<img width="894" alt="Trataasle parans" src="https://github.com/Lightning-AI/lightning/assets/17687676/5afad91e-458f-45f7-a379-f77dea64002a">
The training processing will freeze and no error will be reported as shown in this figure.  After about 30 minutes, start running the "training_step".

**Reply by jxtngx** [reply]
are you also using TensorboardLogger? I am curious if this is an issue with the Tensorboard backend in a distributed training scenario.

I have some requests to help us troubleshoot the problem.

1. the simplest troubleshooting step is to set the logger to None, as xugaoqi1993 did, and then attempt a run
2. if # 1 works, could you please try a run with [CSVLogger](https://lightning.ai/docs/pytorch/stable/extensions/generated/lightning.pytorch.loggers.CSVLogger.html)
3. after 2, please try your current logger again. however, please set `devices` and `num_nodes` to 1, and `strategy` to auto. 

If neither your current logger or CSVLogger work when `devices` >= 2, then it is possible that there is a bug in the way metrics are being collected back from the devices. If CSVLogger works when `devices` >= 2, and not your current logger, then we know it is the current logger. If your current logger works on a single device but not when `devices` >= 2, then we know it is the current logger in a multi-GPU environment.

**Reply by jxtngx** [reply]
after troubleshooting with the above, I will submit an issue referencing this thread.

**Reply by CindyZJT** [reply]
I'm using TensorboardLogger. 
I've followed step 1 to set the logger to None, and the issuse remains.

**Reply by jxtngx** [reply]
Okay.  It is possibly something besides the logger then. Do you agree?

My next idea is to set callbacks and logger to None, so that we can start to isolate the problem without changing the distributed training flags.

**Reply by CindyZJT** [reply]
Thanks for the suggestion. This issuse still exists after setting callbacks and logger to None respectively.

**Reply by jxtngx** [reply]
can you please set `devices` to 1, `strategy` to "auto", and `accelerator` to "gpu".

**Reply by donglihe-hub** [reply]
Met the same issue here. The training hangs after setting logger=True. After about 12 mins the training process finally started.

One possible cause for the issue could be that the hyperparameters to be saved is too large. In my case, the hyperparameter file is 655 MB.

lightning ver.: 2.2.0.post0

```python
    trainer = L.Trainer(
        logger=True,
        num_sanity_val_steps=0,
        accelerator="gpu",
        devices=1,
        enable_progress_bar=True,
        max_epochs=epochs,
        callbacks=callbacks,
        limit_train_batches=limit_train_batches,
        limit_val_batches=limit_val_batches,
        limit_test_batches=limit_test_batches,
        deterministic="warn",
        gradient_clip_val=0.5,
        gradient_clip_algorithm="value",
    )
```
